pkg/storage/edv/restprovider.go

```diff
@@ -516,6 +540,36 @@ func (c *restClient) queryVault(vaultID string, query *models.Query) ([]string,
 	return nil, fmt.Errorf(failResponseFromEDVServer, resp.StatusCode, respBytes)
 }
 
+// DeleteDocument sends the EDV server a request to delete the specified document.
+func (c *restClient) DeleteDocument(vaultID, docID string) error {
+	endpoint := fmt.Sprintf("%s/%s/documents/%s", c.edvServerURL, url.PathEscape(vaultID), url.PathEscape(docID))
+
+	req, err := http.NewRequest(http.MethodDelete, endpoint, nil)
+	if err != nil {
+		return fmt.Errorf(failCreateDELETERequest, err)
+	}
+
+	resp, err := c.httpClient.Do(req) //nolint: bodyclose
+	if err != nil {
+		return fmt.Errorf(failSendDELETERequest, err)
+	}
+
+	defer closeReadCloser(resp.Body)
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf(failReadResponseBody, err)
+	}
+
+	logger.Debugf(sendDELETERequestLogMsg, endpoint, resp.StatusCode, respBytes)
+
+	if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusNoContent {
```

Comment:
How can it be StatusNoContent if you are retrieving document before this call?

Reply:
I'll create another PR and remove that `http.StatusNoContent` check. That was added by mistake.